### ISSUE TITLE
document auto-generate features

### ISSUE THREAD

**Issue by kerrizor** [voormedia/rails-erd#162]
Added by #148 the auto-generate feature will, when installed, generate a new ERD whenever `rake db:migrate` is invoked in development.

**Comment by cherimarie** [comment]
![screen shot 2015-10-23 at 1 21 46 pm](https://cloud.githubusercontent.com/assets/2523812/10704057/0d5495e2-7989-11e5-8586-82a6953c8a0e.png)

We can't get this feature to work, on any of three projects we tried it in. So maybe we need documentation before we can write documentation? :)

Also, does documentation for features like this go in the README or the Jekyll site?

**Comment by mintuhouse** [comment]
There a couple of changes that need to be made before it works.

1. In [auto_generate_diagram.rake](https://github.com/voormedia/rails-erd/blob/master/lib/generators/erd/templates/auto_generate_diagram.rake#L5), 
Erd is undefined.
It should be changed to RailsErd.load_tasks

2. And define load_tasks in [rails_erd.rb](https://github.com/voormedia/rails-erd/blob/master/lib/rails_erd.rb)
````
  def self.loaded_tasks=(val); @loaded_tasks = val; end
  def self.loaded_tasks; return @loaded_tasks; end

  def self.load_tasks
    return if(self.loaded_tasks)
    self.loaded_tasks = true

    Dir[File.join(File.dirname(__FILE__), 'tasks', '**/*.rake')].each { |rake| load rake }
  end
````
Note: See [annotate](https://github.com/ctran/annotate_models/blob/a0682a49c0bbf7590e9fa64d7335554eae688877/lib/annotate.rb#L104) for original implementation

3. Also in [auto_generate_diagram.rake](https://github.com/voormedia/rails-erd/blob/master/lib/tasks/auto_generate_diagram.rake#L18), either we need to define a rake task `erd:default` or change that line to ``bundle exec erd`` 

Note:
Configuring can now only be done via .erdconfig. 
Ideal way I suppose would be invoke CLI and allow config as part of template rake task

**Comment by nielspetersen** [comment]
We've listed this issue within the 14th RubyIssues. https://rubyissues.ongoodbits.com/. :gem: 
We keep our fingers crossed!

**Comment by jasonkarns** [comment]
I believe #252 has addressed this issue and can be closed?

**Comment by kerrizor** [comment]
Closing old maintainer tracking issue. If still relevant, please reopen or create a new issue.
